### PR TITLE
Make Prune Keyspaces Re-usable

### PR DESCRIPTION
**Pull request by PrismaPhonic**
This PR moves prune keyspace logic into shared public functions that can be re-used by consumers.

### CODE REVIEW

File: pkg/operator/vitesstopo/prune_keyspaces.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+// PruneKeyspaces will prune keyspaces that exist but shouldn't anymore.
+func PruneKeyspaces(ctx context.Context, p PruneKeyspacesParams) (reconcile.Result, error) {
+	resultBuilder := &results.Builder{}
+
+	// Make a map from keyspace name (as Vitess calls them) back to the keyspace spec.
+	desiredKeyspaces := make(map[string]*planetscalev2.VitessKeyspaceTemplate, len(p.Keyspaces))
+	for i := range p.Keyspaces {
+		keyspace := &p.Keyspaces[i]
+		desiredKeyspaces[keyspace.Name] = keyspace
```
Comment (avaidyanatha) [inline pkg/operator/vitesstopo/prune_keyspaces.go]:
```suggestion
		desiredKeyspaces[keyspace.Name] = &p.Keyspaces[i]
```

Comment (avaidyanatha) [reply pkg/operator/vitesstopo/prune_keyspaces.go]: Nit: Combine these two lines?

File: pkg/operator/vitesstopo/prune_keyspaces.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+	return resultBuilder.Result()
+}
+
+// KeyspacesToPrune returns a list of keyspace candidates for pruning, based on a provided list of keyspaces to consider.
+func KeyspacesToPrune(keyspaceNames []string, desiredKeyspaces map[string]*planetscalev2.VitessKeyspaceTemplate, orphanedKeyspaces map[string]*planetscalev2.OrphanStatus) []string {
+	var candidates []string
+
+	for _, name := range keyspaceNames {
+		if desiredKeyspaces[name] == nil && orphanedKeyspaces[name] == nil {
+			// The cell exists in topo, but not in the VT spec.
```
Comment (avaidyanatha) [inline pkg/operator/vitesstopo/prune_keyspaces.go]: Do you mean keyspace instead of cell?

File: pkg/operator/vitesstopo/prune_keyspaces.go (hunk 1 above)
Comment (PrismaPhonic) [reply pkg/operator/vitesstopo/prune_keyspaces.go]: good idea - I had simply moved the code but I agree that it's an unnecessary extra line.

File: pkg/operator/vitesstopo/prune_keyspaces.go (hunk 2 above)
Comment (PrismaPhonic) [reply pkg/operator/vitesstopo/prune_keyspaces.go]: Yep, good catch. Fixed.

File: pkg/operator/vitesstopo/prune_keyspaces.go (hunk 1 above)
Comment (PrismaPhonic) [reply pkg/operator/vitesstopo/prune_keyspaces.go]: Done.

**Pull request merged**